consul_registry.go

```diff
@@ -62,13 +62,21 @@ func NewConsulRegister(address string, opts ...Option) (registry.Registry, error
 }
 
 // NewConsulRegisterWithConfig create a new registry using consul, with a custom config.
-func NewConsulRegisterWithConfig(config *api.Config) (*consulRegistry, error) {
+func NewConsulRegisterWithConfig(config *api.Config, opts ...Option) (*consulRegistry, error) {
```

Comment:
Can you additional unit test and sample code ?

Reply:
> Can you additional unit test and sample code ?

done

Reply:
> Can you additional unit test and sample code ?

done

Reply:
how to test this?
```
opts ...Option
```